analyzer/src/test/kotlin/managers/CarthageTest.kt

```diff
@@ -0,0 +1,110 @@
+/*
+ * Copyright (C) 2020 Bosch.IO GmbH
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * License-Filename: LICENSE
+ */
+
+package org.ossreviewtoolkit.analyzer.managers
+
+import io.kotest.core.spec.style.WordSpec
+import io.kotest.matchers.nulls.shouldNotBeNull
+import io.kotest.matchers.shouldBe
+import io.kotest.matchers.string.shouldEndWith
+
+import io.mockk.every
+import io.mockk.mockkStatic
+
+import java.io.File
+import java.net.URL
+
+import org.ossreviewtoolkit.utils.test.DEFAULT_ANALYZER_CONFIGURATION
+import org.ossreviewtoolkit.utils.test.DEFAULT_REPOSITORY_CONFIGURATION
+import org.ossreviewtoolkit.utils.test.USER_DIR
+
+class CarthageTest : WordSpec() {
+
+    private val carthage =
+        Carthage("Carthage", USER_DIR, DEFAULT_ANALYZER_CONFIGURATION, DEFAULT_REPOSITORY_CONFIGURATION)
+
+    init {
+        "resolveDependencies" should {
+            "parse a github dependency" {
+                val cartfile = File("src/test/assets/carthage/Cartfile-github-single.resolved")
+                val result = carthage.resolveDependencies(cartfile).single()
+                result.apply {
+                    packages.size shouldBe 1
+                    packages.forEach {
+                        it.id.type shouldBe "Carthage"
+
+                        it.vcs.url shouldBe "https://github.com/Alamofire/AlamofireImage.git"
+                        it.vcs.revision shouldBe "3.2.0"
+
+                        it.sourceArtifact.url.shouldNotBeNull()
+                        it.sourceArtifact.url shouldEndWith it.vcs.revision
+                    }
+                }
+            }
+
+            "parse a generic git dependency" {
+                val cartfile = File("src/test/assets/carthage/Cartfile-generic-git-single.resolved")
+                val result = carthage.resolveDependencies(cartfile).single()
+                result.packages.apply {
+                    size shouldBe 1
+                    forEach {
+                        it.id.type shouldBe "Carthage"
+                        it.vcs.url shouldBe "https://host.tld/path/to/project.git"
+                        it.vcs.revision shouldBe "1.0.0"
+                    }
+                }
+            }
+
+            "parse a binary dependency url" {
+                mockkStatic("kotlin.io.TextStreamsKt")
```

Comment:
Would it make sense to expose a static `resolveBinaryDependencyUrl(file: String): String?`, and then assert its return values instead of going via the result?

Reply:
Not sure what exactly you mean.

`resolveBinaryDependencyUrl(file: String): String?`
I assume `file` would be the content of the binary specification file/url (or the `file://` or `https://` url) and the return value would be the zip URL to the actual dependency. Then this method would also require the version of the binary dependency.

And then you would mock this `resolveBinaryDependencyUrl` instead of mocking the `URL().readText()` call

Is that correct?

Reply:
Yeah, I think this is what I meant. However, I missed that the method requires also the version. 
Probably it's better to just keep it `as-is`.